boshio/boshio.go

```diff
@@ -245,3 +274,49 @@ func (c Client) retryableRequest(stemcellURL string, byteRange string) ([]byte,
 		return respBytes, nil
 	}
 }
+
+func (c Client) fetchWithAuth(urlString string, bytes int, offset int, auth Auth) ([]byte, error) {
+	reader, err := c.minioReaderForObject(urlString, auth)
+	if err != nil {
+		return nil, err
+	}
+
+	byteSegment := make([]byte, bytes)
+	reader.ReadAt(byteSegment, int64(offset))
```

Comment:
Any reason not to handle the error by `ReadAt`?

Reply:
Looks like an oversight. Added the nil check now.

Thanks!